File: starterkit/c-sharp-unity/__PREFIX__Control.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public partial class __PREFIX__Control : MonoBehaviour {
 
    #region manger
    Action<bool> m_curfunc;
    Action<bool> m_nextfunc;
    Action<bool> m_tempfunc;

    bool         m_noWait;
    
    void _update()
    {
        while(true)
        {
            var bFirst = false;
            if (m_nextfunc!=null)
            {
                m_curfunc = m_nextfunc;
                m_nextfunc = null;
                bFirst = true;
            }
            m_noWait = false;
            if (m_curfunc!=null)
            {   
                m_curfunc(bFirst);
            }
            if (!m_noWait) break;
        }
    }
    void Goto(Action<bool> func)
    {
        m_nextfunc = func;
    }
    bool CheckState(Action<bool> func)
    {
        return m_curfunc == func;
    }
    // for tempfunc
    void SetNextState(Action<bool> func)
    {
        m_tempfunc = func;
    }
    void GoNextState()
    {
        m_nextfunc = m_tempfunc;
        m_tempfunc = null;
    }
    bool HasNextState()
    {
        return m_tempfunc != null;
    }
    void NoWait()
    {
        m_noWait = true;
    }
    #endregion

    void _start()
    {
        Goto(S_START);
    }
    public bool IsEnd()     
    { 
        return CheckState(S_END); 
    }

	#region    // [SYN-G-GEN OUTPUT START] indent(8) $/./$
    void S_START(bool bFirst) { }
    void S_END(bool bFirst) { }
	#endregion // [SYN-G-GEN OUTPUT END]

	// write your code below

	bool m_bYesNo;
	
	void br_YES(Action<bool> st)
	{
		if (!HasNextState())
		{
			if (m_bYesNo)
			{
				SetNextState(st);
			}
		}
	}

	void br_NO(Action<bool> st)
	{
		if (!HasNextState())
		{
			if (!m_bYesNo)
			{
				SetNextState(st);
			}
		}
	}

    #region Monobehaviour framework
    void Start()
    {
        _start();
    }
    void Update()
    {
        if (!IsEnd()) 
        {
            _update();
        }
    }
    #endregion
}

File: starterkit/visual-basic/__PREFIX__Control.vb
Public Class __PREFIX__Control

    ' Manager
    Private m_curstate As String
    Private m_nextstate As String
    Private m_nowait As Boolean

    Public Sub New()
        m_curstate = ""
        m_nextstate = ""
    End Sub

    Public Sub Update()
        Dim bFirst As Boolean

        LoopUpdate:
        
        bFirst = False

        If m_nextstate <> "" Then 
            m_curstate = m_nextstate
            m_nextstate = ""
            bFirst = True
        End If

        m_nowait = False

        '[SYN-G-GEN OUTPUT START] indent(8) $/^S_/->#select$
        If m_curstate = "S_START" Then 
            S_START(bFirst)
            Goto ExitUpdate
        End If
        If m_curstate = "S_END" Then 
            S_END(bFirst)
            Goto ExitUpdate
        End If
        '[SYN-G-GEN OUTPUT END]

        ExitUpdate:

        If m_nowait = True Then
            Goto LoopUpdate
        End If

    End Sub

    private Sub GotoState( state As String )
        m_nextstate = state
    End Sub

    private Function CheckState( state As String ) As Boolean
        If state = m_curstate Then
            Return True
        Else
            Return False
        End If
    End Function

    private Function HasNextState()
        If m_nextstate <> "" Then
            Return True
        Else
            Return False
        End If
    End Function

    private Sub SetNoWait()
        m_nowait = True
    End Sub

    public Sub Start()
        GotoState("S_START")
    End Sub

    public Function IsEnd() As Boolean
        If m_curstate = "S_END" Then
            Return True
        Else
            Return False
        End If
    End Function

    '[SYN-G-GEN OUTPUT START] indent(8) $/./$
    private Sub S_START( bFirst As Boolean )
        GotoState("S_END")
    End Sub
    private Sub S_END( bFirst As Boolean )
        '
    End Sub
    '[SYN-G-GEN OUTPUT END]

End Class